packages/core/src/controllers/subscriber.ts

```diff
@@ -401,4 +410,16 @@ export class Subscriber extends ISubscriber {
       throw new Error(message);
     }
   }
+
+  private async restartToComplete() {
+    if (this.restartInProgress) return;
+
+    await new Promise<void>((resolve) => {
+      setInterval(() => {
+        if (!this.restartInProgress) {
+          resolve();
```

Comment:
Think we need to call `clearInterval(id)` here before resolving to avoid possible memory leakage from zombie intervals: https://developer.mozilla.org/en-US/docs/Web/API/setInterval#return_value

Reply:
I thought to add it & forgot 🤦